cpp/include/cuml/fil/multi_reduction.h

```diff
@@ -0,0 +1,31 @@
+#pragma once
+#include <raft/cuda_utils.cuh>
+// data [T]ype, reduction [R]adix
+template <int R, typename T>
```

Comment:
I'm actually not sure what order is the best:
- `T` as the first argument allows the default value for `R`: `template <typename T, int R = 4>`;
- `R` as the first parameter allows specifying the radix while deducing `T`.

The decision is up to you.